irony-cdb.el

```diff
@@ -63,6 +63,20 @@ for files that it cannot handle."
   :type '(repeat function)
   :group 'irony-cdb)
 
+(defcustom irony-cdb-search-directory-list '("." "build")
```

Comment:
> 3) and the user actually wants to use another cdb, which isn't located in either the same directory or a parent directory.
> 
> ....? and you believe that situation affects more users then those who use CMake and follow the build/ convention for which now irony will work out of the box?

I don't believe this, I believe it will happen mostly to me.


> I would actually favor the opposite approach. Let's add a check to verify (buffer-file-name) is included in the cdb before we accept it and then be very permissive about where we look by default. If there are any semi-popular conventions on where the build directory lives, just throw'em all in there.

Let's keep it simple and NOT do that for now. If someone encounter the issue we will discuss it then.